app/scripts/modules/google/src/interceptors/iap.interceptor.ts

```diff
@@ -0,0 +1,31 @@
+import { IHttpInterceptor, IHttpPromiseCallbackArg, IHttpProvider, IPromise, module } from 'angular';
+import { $http, $q } from 'ngimport';
+import { SETTINGS } from '@spinnaker/core';
+
+/**
+ * When a Cloud IAP (if enabled) session expires, refreshes session.
+ */
+
+export class IapInterceptor implements IHttpInterceptor {
+  public responseError = <T>(response: IHttpPromiseCallbackArg<T>): IPromise<T> => {
+    const { config, status } = response;
+
+    if (status === 401 && SETTINGS.feature.iapRefresherEnabled) {
+      return $http.get('/_gcp_iap/do_session_refresh').then(
+        () => {
+          return $q.resolve($http.get(config.url, config)).then((result: T) => result, () => $q.reject(response));
```

Comment:
`$http.get(...)` returns a promise already - so what you have here is equivalent to just `$http.get(config.url, config)`